frontend/fingertips-frontend/app/chart/PopulationPyramidWithTableDataProvider/PopulationPyramidWithTableDataProvider.test.tsx

```
@@ -83,32 +82,61 @@ describe('PopulationPyramidWithTableDataProvider', () => {
     const view = render(jsxView);
     expect(view).toBeTruthy();
     expect(view.getByText('PopulationPyramidWithTable')).toBeInTheDocument();
-    expect(mockGetArea).toHaveBeenCalledTimes(1);
+    expect(mockGetArea).toHaveBeenCalledTimes(2);
     expect(mockGetHealthDataForAnIndicator).toHaveBeenCalledTimes(1);
     expect(mockIndicatorSearchService.getIndicator).toHaveBeenCalledTimes(1);
   });
 
   it('renders PopulationPyramidWithTable with more than 100 areas data to be fetched', async () => {
-    const areaCodes = ((n: number) => {
+    const areas = ((n: number) => {
       //generate random area codes
-      const results: string[] = [];
+      const results: Area[] = [];
       for (let i = 0; i < n; i++) {
         const randomNum = Math.floor(Math.random() * 100000);
         const paddedNum = String(randomNum).padStart(5, '0');
         const randomString = `E090${paddedNum}`;
-        results.push(randomString);
+        const area = {
+          areaType: {
+            hierarchyName: HierarchyNameTypes.NHS,
+            key: '',
+            name: 'NHS something',
+            level: 0,
+          },
+          code: randomString,
+          name: 'Test Area',
+          parent: {
+            key: '',
+            level: 1,
+            name: 'Test Area',
+            areaType: { hierarchyName: HierarchyNameTypes.NHS },
+          },
+        };
+        results.push(area);
       }
       return results;
     })(130);
 
+    mockGetArea.mockImplementation(({ areaCode }: { areaCode: string }) => {
+      const area = areas.find((area) => area.code === areaCode);
+      if (!area) {
```

is this not masking failures and making the test non-deterministic?